cf-revision-manager.php

```diff
@@ -2,200 +2,321 @@
+*
+* @since 1.0.1
+*
+* @return mixed Returns the global value depending on the key being provided
 */
-if (!class_exists('cf_revisions')) {
-	
-	define('CF_REVISIONS_DEBUG', false);
-	
-	function cfr_register_metadata($postmeta_key, $display_func = '') {
-		static $cfr;
-		if (empty($cfr)) {
-			$cfr = cf_revisions::get_instance();
-		} 
-		return $cfr->register($postmeta_key, $display_func);
+function cfrm_get_global( $key ) {
+	$cfrm                   = array();
+	$cfrm['plugin_file']    = __FILE__;
+	$cfrm['basename']       = plugin_basename( __FILE__ );
+	$cfrm['directory_path'] = plugin_dir_path( __FILE__ );
+	$cfrm['directory_url']  = plugin_dir_url( __FILE__ );
+	return array_key_exists( $key, $cfrm ) ? $cfrm[$key] : '';
+}
+
+/**
+ * Register js and css file.
+ *
+ * @since 1.0.1
+ */
+function cfrm_enqueue_js() {
+	wp_register_script( 'cfrm-script', cfrm_get_global( 'directory_url' ) . 'cfrm-script.js', array( 'jquery' ) );
+	wp_register_style( 'cfrm-style', cfrm_get_global( 'directory_url' ) . 'cfrm-style.css' );
+}
+add_action( 'admin_enqueue_scripts', 'cfrm_enqueue_js' );
+
+/**
+ * Enqueue js and css file in revision page.
+ *
+ * @since  1.0.1
+ * @return void
+ */
+function cfrm_admin_head() {
+	if ( ! empty( $_GET['page'] ) && $_GET['page'] == basename( __FILE__ ) ) {
+		wp_enqueue_style( 'cfrm-style' );
+		wp_enqueue_script( 'cfrm-script' );
 	}
+}
+add_action( 'admin_head', 'cfrm_admin_head' );
+
+/**
+ * Register revision manger admin menu.
+ *
+ * @since  1.0.1
+ * @return void
+ */
+function cfrm_admin_menu() {
+	 add_options_page(
+        __( 'Revision Manager', 'cfrm' ),
+        __( 'Revision Manager', 'cfrm' ),
+        'manage_options',
+        basename( __FILE__ ),
+        'cfrm_admin_form'
+    );
+}
+add_action( 'admin_menu', 'cfrm_admin_menu' );
 
-	class cf_revisions {
-		private static $_instance;
-		protected $postmeta_keys = array();
-	
-		public function __construct() {
-			# save & restore
-			add_action('save_post', array($this, 'save_post_revision'), 10, 2);
-			add_action('wp_restore_post_revision', array($this, 'restore_post_revision'), 10, 2);
-
-			if (is_admin()) {		
-				# revision display
-				global $pagenow;
-				if ($pagenow == 'revision.php') {
-					add_filter('_wp_post_revision_fields', array($this, 'post_revision_fields'), 10, 1);
-					add_filter('_wp_post_revision_field_postmeta', array($this, 'post_revision_field'), 1, 2);
+/**
+ * Revision Manger admin form callback function.
+ *
+ * @since  1.0.1
+ * @return void
+ */
+function cfrm_admin_form() {
+	//include_once cfase_get_global( 'directory_path' ) . 'includes/admin-form.php';
+	global $CFR_KEYS_REQUIRED;
+	$required_keys = $CFR_KEYS_REQUIRED;
+	$keys = array_diff( cfrm_meta_keys(), excluded_keys(), $required_keys );
+?>
+<div class="wrap">
+	<h2><?php _e( 'CF Revision Manager', 'cfrm' ); ?></h2>
+<?php
+		if ( ! count( $keys ) ) {
+			echo '<p>'.__( 'No custom fields found.', 'cfrm' ).'</p>';
```

Comment:
there should be a space on each side of both concatenating periods here:
```php
echo '<p>'.__( 'No custom fields found.', 'cfrm' ).'</p>';
```
becomes
```php
echo '<p>' . __( 'No custom fields found.', 'cfrm' ) . '</p>';
```